libsolidity/formal/CHC.cpp

```diff
@@ -164,11 +164,11 @@ void CHC::endVisit(ContractDefinition const& _contract)
 		if (base != &_contract)
 		{
 			m_callGraph[&_contract].insert(base);
-			vector<ASTPointer<Expression>> const& args = baseArgs.count(base) ? baseArgs.at(base) : decltype(args){};
 
 			auto baseConstructor = base->constructor();
-			if (baseConstructor && !args.empty())
+			if (baseConstructor && baseArgs.count(base))
```

Comment:
I guess that works. I'm wondering if it's possible that `base` is in `baseArgs` but with empty args, in which case it shouldn't go in here. Well, I guess we'll see via the tests

Reply:
I was wondering the same... I had a different solution earlier that had zero change in behaviour, but this one would be simpler...
I mean the only question is if in that case, i.e. ``base`` is in ``baseArgs``, but with empty args, we can assume that ``baseConstructor->parameters().size() == 0`` as well, for the assertion below to hold, right? If it does, then entering here is not a problem...

Reply:
But ``base`` will only be in ``baseArgs``, if that inheritance specifier has ``arguments() != 0``, so it's ``Base()``, not ``Base`` - and that's only valid, if the base constructor takes no arguments, so it should be fine, right :-)?
If you think that's fine, we can keep it - if we're in doubt I can just make it a pointer, then it's exactly the same as before.

Reply:
Tests also seem fine.

Reply:
Yea, that's also what I thought, should be correct

Reply:
Does that mean approval or not ;-)?

Reply:
Does it though?

Reply:
Apparently it does, doesn't it :-)?

Reply:
I guess it does, does it not?

Reply:
Well now it is, is it not?

Reply:
Looks like it is, doesn't it?

Reply:
I wouldn't doubt it, would I?

Reply:
I guess I'll just do it then, won't I?